preliz/ppls/pymc_io.py

```diff
@@ -201,11 +200,54 @@ def reshape_params(model, var_info, p_model, params):
     return value
 
 
-def non_constant_parents(var_):
+def non_constant_parents(var_, free_rvs):
     """Find the parents of a variable that are not constant."""
     parents = []
     for variable in var_.get_parents()[0].inputs[2:]:
         if not isinstance(variable, TensorConstant):
-            parents.append(variable.owner.inputs[0])
-
+            for free_rv in free_rvs:
+                if free_rv in list(ancestors([variable])) and free_rv not in parents:
+                    parents.append(free_rv)
     return parents
+
+
+def parse_backfitting(model, data, alternative=None):
+    """
+    Get the model information for fitting the samples from prior into user provided model's prior.
+    We need to "backfit" because we can not use arbitrary samples as priors.
+
+    Parameters
+    ----------
+    model : A PyMC model
+    A probabilistic model
+
+    data : np.ndarray
+    Samples which needs to be fitted using "backfitting"
+
+    alternative : list, defaults to None
+    Users can add the model variables to consider alternative distributions while fitting samples
+
+    """
+
+    data = np.asarray(data)
+    with model:
+        idata = sample(random_seed=4124)
+    posterior = az.extract(idata, group="posterior")
+    model_info = get_model_information(model)[2]
+    if alternative:
```

Comment:
Not sure I fully understand the logic inside the alternative branch. It should be pretty similar to the regular branch except that you should use `pz.mle` instead of  `dist._fit_mle`. Besides with the current logic you are fitting the distributions twice, once inside the alternative branch and then again when you evaluate the distributions inside  `parsed_info`.

Reply:
Have removed `dist._fit_mle` and replaced it with `pz.mle`